DorisProject/src/app/pages/add-category/add-category.component.ts

```diff
@@ -32,22 +41,30 @@ export class AddCategoryComponent implements OnInit {
   save() {
     this.mostrar = false;
     this.mostrarError = false;
-    const category = { "name": this.addCategoryForm.get('name').value, "description": this.addCategoryForm.get('description').value, "active": "Y" };
+    const category = {
+      'name': this.addCategoryForm.get('name').value,
+      'description': this.addCategoryForm.get('description').value,
+      'active': 'Y'
+    };
     this.categoryService.save(category).subscribe(data => {
-      this.mensaje = "Category created successfully!!";
-      this.mostrar = true;
-      this.addCategoryForm.reset();           
-    }, err => { 
-      this.mensaje = err.error.text;
-      this.mostrarError = true;
-    }
+        this.mensaje = 'Category created successfully!!';
+        this.openModal('succes');
+        this.addCategoryForm.reset();
+      }, err => {
+
+        if (err.error.text === 'Category name is exist') {
+          this.openModal('errorExitsCategoryy');
+        }else {
```

Comment:
Codacy has a fix for the issue: [Insert `·`](https://app.codacy.com/gh/SaraArenasV/DorisProjectFrontEnd/pullRequest?prid=9258681)
```suggestion
        } else {
```